StrongInject.Tests.Unit/OwnedInjectionTests.cs

```diff
@@ -0,0 +1,2042 @@
+public partial class Container : IContainer<Owned<A>> { }
+
+public record A : IDisposable { public void Dispose() { } };
+";
+            var comp = RunGeneratorWithStrongInjectReference(userSource, out var generatorDiagnostics, out var generated);
+            generatorDiagnostics.Verify();
+            comp.GetDiagnostics().Verify();
+            var file = Assert.Single(generated);
+            file.Should().BeIgnoringLineEndings(@"#pragma warning disable CS1998
+partial class Container
+{
+    private int _disposed = 0;
+    private bool Disposed => _disposed != 0;
+    public void Dispose()
+    {
+        var disposed = global::System.Threading.Interlocked.Exchange(ref this._disposed, 1);
+        if (disposed != 0)
+            return;
+    }
+
+    TResult global::StrongInject.IContainer<global::StrongInject.Owned<global::A>>.Run<TResult, TParam>(global::System.Func<global::StrongInject.Owned<global::A>, TParam, TResult> func, TParam param)
+    {
+        if (Disposed)
+            throw new global::System.ObjectDisposedException(nameof(Container));
+        global::StrongInject.Owned<global::A> owned_0_0;
+        global::StrongInject.Owned<global::A> CreateOwnedA_1()
+        {
+            global::A a_0_0;
+            a_0_0 = new global::A();
+            return new global::StrongInject.Owned<global::A>(a_0_0, () =>
+            {
+                ((global::System.IDisposable)a_0_0).Dispose();
+            });
+        }
+
+        owned_0_0 = CreateOwnedA_1();
+        TResult result;
+        try
+        {
+            result = func(owned_0_0, param);
+        }
+        finally
+        {
+        }
+
+        return result;
+    }
+
+    global::StrongInject.Owned<global::StrongInject.Owned<global::A>> global::StrongInject.IContainer<global::StrongInject.Owned<global::A>>.Resolve()
+    {
+        if (Disposed)
+            throw new global::System.ObjectDisposedException(nameof(Container));
+        global::StrongInject.Owned<global::A> owned_0_0;
+        global::StrongInject.Owned<global::A> CreateOwnedA_1()
+        {
+            global::A a_0_0;
+            a_0_0 = new global::A();
+            return new global::StrongInject.Owned<global::A>(a_0_0, () =>
+            {
+                ((global::System.IDisposable)a_0_0).Dispose();
+            });
+        }
+
+        owned_0_0 = CreateOwnedA_1();
+        return new global::StrongInject.Owned<global::StrongInject.Owned<global::A>>(owned_0_0, () =>
+        {
+        });
+    }
+}");
+        }
+
+        [Fact]
+        public void OwnedCanBeManuallyProvidedViaFactory()
+        {
+            string userSource = @"
+using System;
+using StrongInject;
+
+[Register(typeof(A)), Register(typeof(B))]
+public partial class Container : IContainer<A>
+{
+    [Factory]
+    public static Owned<B> CreateOwnedManually() => throw new NotImplementedException();
+}
+
+public record A(Owned<B> B);
+public record B : IDisposable { public void Dispose() { } }
+";
```

Comment:
Re: https://github.com/YairHalberstadt/stronginject/pull/143#discussion_r700759638

> We should ideally handle the case where somebody registers a `[Factory] T GetAnything<T>` and not use it if `T` is an `Owned<T>`. I think the most sensible option for me would probably be to just disallow that completely, which would be a breaking change, but I realize that there's almost no way to use that safely. The reason I allowed it is because maybe you want a last chance provider which tries to create it via reflection, or by calling another container, but that just doesn't seem sensible. So no need for you to handle that yourself - I'm happy to do so.

This isn't what I expected. Why wouldn't there be a safe way to return `Owned<X>` or `AsyncOwned<X>` from a `[Factory]`?

Reply:
I didn't understand at first, but I just realized what you're saying. An open generic factory method that doesn't specifically return `Owned<T>` would still be used because StrongInject would parameterize the method with `T = Owned<X>` at some point. And you're saying that is future work outside the scope of this PR since such a method is already problematic everywhere. So, this is another "possible future PR" comment. :+1: